### PR TITLE
op-deployer: Support forking live chains

### LINKED ISSUE #12855: Add support for forking an existing chain
This will allow us to unify code paths in the deployment pipeline that use predeployed contracts.

### PR DESCRIPTION
**Pull request by mslipper**
Integrates op-deployer with the forking functionality added in https://github.com/ethereum-optimism/optimism/pull/11919. With this PR, op-deployer can successfully simulate transactions against a forked remote chain. This dramatically simplifies the deployment pipeline, since there's no need to re-inject state at every step or manually craft calldata during the OP Chain deployment step.

As part of this PR, I simplified how the implementations get set post-OP Chain deployment in the event an existing OPCM is used. I created a `ReadImplementationAddresses` script that pulls the implementations from the proxies in Solidity, then use the scripting functionality in Go to call the script. This eliminated the need to perform lots of RPC calls and manually decode the results.

Since op-deployer supports deploying v1.6.0, I had to backport the `ReadImplementationAddresses` script to v1.6.0. You can see the diff between the base v1.6.0 release [here](https://github.com/ethereum-optimism/optimism/commit/5d158f53803e958c004e32bff7ab2d4bc3270211). I gave it the tag [op-contracts-v160-artifacts-read-implementations-backport](https://github.com/ethereum-optimism/optimism/releases/tag/op-contracts-v160-artifacts-read-implementations-backport) so it can be easily found in the future. I also tagged the base v1.6.0 release as [op-contracts-v160-artifacts-base](https://github.com/ethereum-optimism/optimism/releases/tag/op-contracts-v160-artifacts-base) for the same reason.

Lastly, I updated the artifacts publishing script to include the Git commit the artifacts were generated from to make determining where a given artifacts bundle was generated from easier in the future.

Closes https://github.com/ethereum-optimism/optimism/issues/12855.

### CODE REVIEW

File: op-deployer/pkg/deployer/apply.go [hunk 1]
```diff
@@ -175,52 +172,101 @@ func Apply(ctx context.Context, cfg ApplyConfig) error {
...
-	env *pipeline.Env,
-	bundle pipeline.ArtifactsBundle,
-	intent *state.Intent,
-	st *state.State,
-) error {
 	pline := []pipelineStage{
 		{"init", func() error {
 			if intent.DeploymentStrategy == state.DeploymentStrategyLive {
-				return pipeline.InitLiveStrategy(ctx, env, intent, st)
+				return pipeline.InitLiveStrategy(ctx, pEnv, intent, st)
```
Comment (blmalone) [inline op-deployer/pkg/deployer/apply.go:260]: dumb q, what's `pEnv` stand for over `env`?

File: packages/contracts-bedrock/scripts/deploy/ReadImplementationAddresses.s.sol [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            vm.prank(address(0));
+            _rio.set(sels[i], proxy.implementation());
+        }
+
+        vm.prank(address(0));
+        address l1SBImpl = IStaticL1ChugSplashProxy(address(_rii.l1StandardBridgeProxy())).getImplementation();
+        vm.prank(address(0));
+        _rio.set(_rio.l1StandardBridge.selector, l1SBImpl);
+
+        (address mipsLogic,) = _rii.opcmProxy().implementations(_rii.release(), "MIPS");
```
Comment (blmalone) [inline packages/contracts-bedrock/scripts/deploy/ReadImplementationAddresses.s.sol:157]: I've added getters in the new OPCM redesign. https://github.com/ethereum-optimism/optimism/pull/12851

File: op-deployer/pkg/deployer/apply.go (hunk 1 above)
Comment (mslipper) [reply op-deployer/pkg/deployer/apply.go:260]: pipelineEnv - I renamed it so it wouldn't shadow the `env` package.

File: packages/contracts-bedrock/scripts/deploy/ReadImplementationAddresses.s.sol (hunk 2 above)
Comment (mslipper) [reply packages/contracts-bedrock/scripts/deploy/ReadImplementationAddresses.s.sol:157]: Nice, that'll clean this up a lot.

**Review by blmalone — APPROVED** [review]
Awesome!

**Pull request merged**